src/derivative_utils.jl

```diff
@@ -531,21 +531,27 @@ function build_J_W(alg,u,uprev,p,t,dt,f,uEltypeNoUnits,::Val{IIP}) where IIP
     end
     W = WOperator(f.mass_matrix, dt, J, IIP)
   else
-    J = false .* _vec(u) .* _vec(u)'
+    J = if f.jac_prototype === nothing
+      false .* _vec(u) .* _vec(u)'
+    else
+      deepcopy(f.jac_prototype)
+    end
     isdae = alg isa DAEAlgorithm
     W = if isdae
       J
     elseif IIP
       similar(J)
     else
-      W = if u isa StaticArray
-      lu(J)
+      if u isa StaticArray
```

Comment:
Those branches can be removed with `lu_instance`.

Reply:
Done